diff --git a/android/build.gradle b/android/build.gradle
index cecc20dc722556e93a840ac277886ff92111e4a3..93a772d55f38d62c4569792a3cca64dbbc73e3e2 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -121,6 +121,8 @@ def resolveReactNativeDirectory() {
         return reactNativeFromNodeModulesWithImagePicker
     }
 
+    return file("${projectDir}/../../../../../node_modules/react-native")
+
     throw new Exception(
             "[react-native-image-picker] Unable to resolve react-native location in " +
                     "node_modules. You should add project extension property (in app/build.gradle) " +
